tools/wpt/virtualenv.py

```diff
@@ -8,18 +8,40 @@
 
 logger = logging.getLogger(__name__)
 
+
+def add_site_package_path(root_path):
+    """Update system path to include packages from existing virtualenv"""
+    for root, dirs, files in os.walk(root_path + "/lib"):
+        if "site-packages" in dirs:
+            site_package_path = root + "/site-packages"
+            logger.info("Using site packages from %s" % site_package_path)
+            sys.path.append(site_package_path)
+            return
+
+
 class Virtualenv(object):
-    def __init__(self, path):
+    def __init__(self, path, skip_virtualenv_setup):
         self.path = path
-        self.virtualenv = find_executable("virtualenv")
-        if not self.virtualenv:
-            raise ValueError("virtualenv must be installed and on the PATH")
+        self.skip_virtualenv_setup = skip_virtualenv_setup
+        if skip_virtualenv_setup:
```

Comment:
My fault, I didn't realize that the activation script does the path adjustment, so this can be removed.

In my case, though, the virtualenv has already been activated by the time WPT runs, so the activation script is not there. How about we try to run the activation script, and if it fails (likely because the file doesn't exist), we ignore the error when `skip_virtualenv_setup` is True.

Reply:
Eh? The file does still exist when the virtualenv has been activated. Are you actually running something that isn't really virtualenv and so behaves slightly differently?

Reply:
So on my end, there is a virtualenv on the machine but it's in a separate location from where WPT lives, and $PATH is already updated with the right directories. So I should actually be able to locate this file via $PATH, but I don't need to because it's already been activated. So I don't actually need any of this path updating logic, sorry for the churn.

In general it seems safe to assume that the virtualenv has been activated if we're skipping its setup. Let me know if you disagree.